webvtt/writers.py

```diff
@@ -2,12 +2,20 @@
 class WebVTTWriter(object):
 
     def write(self, captions, f):
-        f.write('WEBVTT\n')
-        for c in captions:
-            if c.identifier:
-                f.write('\n' + c.identifier)
-            f.write('\n{} --> {}\n'.format(c.start, c.end))
-            f.writelines(['{}\n'.format(l) for l in c.lines])
+        f.write(self.webvtt_content(captions))
+
+    def webvtt_content(self, captions):
```

Comment:
To do this better create a list and add each line to it, then use '\n'.join.

Reply:
@glut23 If each line were separated by a single `\n`, it would have made sense. But there is inconsistency in the newline after WEBVTT, identifier, and timestamps. Using a list would make it a bit difficult to read and understand because of the combination of strings with \n or empty strings during appending and the last join to get the desired results. Adding one variant of code as following:

```
        output = ["WEBVTT"]
        for caption in captions:
            if caption.identifier:
                output.append("\n{}".format(caption.identifier))
            output.append('\n{} --> {}'.format(caption.start, caption.end))
            for line in caption.lines:
                output.append(line)
        return '\n'.join(output)
```

Reply:
Hi @DawoudSheraz apologies for the late reply. That code makes the current tests to break. The \n handling in between elements is not very readable so I would just get rid of it. This is how I would do it:


        output = ['WEBVTT']
        for caption in captions:
            output.append('')
            if caption.identifier:
                output.append(caption.identifier)
            output.append('{} --> {}'.format(caption.start, caption.end))
            output.extend(caption.lines)
        return '\n'.join(output)



Also you test had an issue expecting a last blank line to be present in the resulting content. You can do this:


        expected_content = textwrap.dedent("""
                WEBVTT

                00:00:00.500 --> 00:00:07.000
                Caption test line 1
                Caption test line 2

                00:00:08.000 --> 00:00:15.000
                Caption test line 3
                Caption test line 4
                """).strip()